sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -971,6 +971,10 @@ object QueryExecutionErrors {
     new DateTimeException(s"Cannot cast $s to $to.")
   }
 
+  def cannotCastDoubleToTimestampError(d: Double, to: DataType): Throwable = {
```

Comment:
Instead of adding a new error here, how about merge it with the `cannotCastUTF8StringToDataTypeError` and maybe call it `cannotCastToDataTypeError`? Since they have the same error message and exception type.

Reply:
+1

Reply:
Updated